### PR TITLE
added definitions for redux-subscriber

### PR DESCRIPTION
**Pull request by thisissami**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by thisissami** [comment]
Ok, I just looked through everything - and everything checks off! For what it's worth, the package maintainer has an `index.d.ts` in their source code, but 1) it isn't part of the npm package, and 2) it isn't very good (it has lots of `any`s and doesn't actually match the shape of the module). Accordingly, I'm submitting as part of this PR. :)

Let me know if I did anything wrong!

**Comment by thisissami** [comment]
And I just ran one last set of functions in the test to have full API coverage. :)

**Comment by peterblazejewicz** [comment]
@thisissami can you verify with package owner what is the intent of the already xist definitiion file for this package?:
https://github.com/ivantsov/redux-subscriber/blob/master/index.d.ts
It seems it was never published to NPM and `package.json` is not modified to denote this:
https://betterstack.dev/blog/publishing-type-definitions-with-npm-package/#heading-including-type-definitions

**Comment by thisissami** [comment]
Hi @peterblazejewicz - I can open an issue and tag them in it, but I have no idea if they'll see/respond to my request. This package hasn't been updated in 2 years. I can tell you that the definition file in the package itself is lacking (which is why I wrote my own that accurately reflects the library's API).

**Comment by thisissami** [comment]
@peterblazejewicz just FYI: https://github.com/ivantsov/redux-subscriber/issues/14#issuecomment-626473573

File: types/redux-subscriber/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Type definitions for redux-subscriber 1.1
+// Project: https://github.com/ivantsov/redux-subscriber#readme
+// Definitions by: thisissami\n <https://github.com/thisissami>
```
Comment (peterblazejewicz) [inline types/redux-subscriber/index.d.ts:3]:
```suggestion
// Definitions by: thisissami <https://github.com/thisissami>
```

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@thisissami thx for the contribution!
I understand we assume that this DT can be retracted depending on the status of the original package, see comment and below:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44534#issuecomment-626300879
Thanks!

**Pull request merged**